src/graphics.cpp

```diff
@@ -105,9 +249,35 @@ void GetvScreen(const int A)
             if(vScreen[A].TempY < 0)
                 vScreen[A].TempY += 1;
         }
-        if(p.Mount == 2)
-            pLoc.Height = 128;
     }
+
+    // keep vScreen boundary even on 3DS
+#ifdef __3DS__
+    vScreenX[A] -= std::fmod(vScreenX[A], 2.);
+    vScreenY[A] -= std::fmod(vScreenY[A], 2.);
+#endif
+}
+
+// NEW: get the vScreen position if it were 800x600, and write the top-left coordinate to (left, top)
+void GetvScreenCanonical(int A, int* left, int* top)
+{
+    auto &p = Player[A];
+    auto &pLoc = p.Location;
+
+    double pHeight = (p.Mount != 2) ? pLoc.Height : 0;
+
+    *left = -pLoc.X + (800 * 0.5) - pLoc.Width / 2.0;
+    *top = -pLoc.Y + (600 * 0.5) - vScreenYOffset - pHeight;
+    *left -= vScreen[A].tempX;
+    *top -= vScreen[A].TempY;
+    if(-(*left) < level[p.Section].X)
+        *left = -level[p.Section].X;
+    else if(-(*left) + 800 > level[p.Section].Width)
+        *left = -(level[p.Section].Width - 800);
+    if(-(*top) < level[p.Section].Y)
```

Comment:
Please add blank line between two different condition branches, they shouldn't be mixed.